### ISSUE TITLE
Bug: `ERR_PACKAGE_PATH_NOT_EXPORTED` with TypeScript 7 RC (`7.0.1-rc`)

### ISSUE THREAD

**Issue by chimurai** [typescript-eslint/typescript-eslint#12445]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
https://github.com/chimurai/typescript-rc-eslint-repro
```

### ESLint Config

```javascript
import js from "@eslint/js";
import globals from "globals";
import tseslint from "typescript-eslint";
import { defineConfig } from "eslint/config";

export default defineConfig([
  { files: ["**/*.{js,mjs,cjs,ts,mts,cts}"], plugins: { js }, extends: ["js/recommended"], languageOptions: { globals: globals.node } },
  tseslint.configs.recommended,
]);
```

### tsconfig

```jsonc

```

### Expected Result

no errors when running: `npx eslint .`

### Actual Result

running `npx eslint .`

```shell
@chimurai ➜ /workspaces/typescript-rc-eslint-repro (main) $ npx eslint .

Oops! Something went wrong! :(

ESLint: 10.5.0

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /workspaces/typescript-rc-eslint-repro/node_modules/typescript/package.json
    at exportsNotFound (node:internal/modules/esm/resolve:314:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:662:9)
    at resolveExports (node:internal/modules/cjs/loader:685:36)
    at Module._findPath (node:internal/modules/cjs/loader:752:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1441:27)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1066:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1071:22)
    at Module._load (node:internal/modules/cjs/loader:1242:25)
    at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
    at Module.require (node:internal/modules/cjs/loader:1556:12)
```

### Additional Info

Reproduction with a vanilla project: https://github.com/chimurai/typescript-rc-eslint-repro
- Included step-by-step instructions in the README.md
- Granular git commits with the same steps-by-step

also reported: https://github.com/microsoft/vscode-eslint/issues/2195

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.61.1` |
| `@typescript-eslint/parser`            | `8.61.1` |
| `@typescript-eslint/rule-tester`       | `not installed` |
| `@typescript-eslint/scope-manager`     | `8.61.1` |
| `@typescript-eslint/typescript-estree` | `8.61.1` |
| `@typescript-eslint/type-utils`        | `8.61.1` |
| `@typescript-eslint/utils`             | `8.61.1` |
| `TypeScript`                           | `7.0.1-rc` |
| `ESLint`                               | `10.5.0` |
| `node`                                 | `24.14.0` |

**Comment by kirkwaiblinger** [comment]
Hi @chimurai! This is expected... TS 7 is not supported by typescript-eslint and cannot be supported for the known future (due to [TS 7 lacking an API](https://github.com/microsoft/typescript-go/discussions/455)). 

However, normally, typescript-eslint should issue a warning when you're using an unsupported TS version, like:

```plaintext
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

* @typescript-eslint/typescript-estree version: 8.15.0
* Supported TypeScript versions: >=4.7.4 <5.7.0
* Your TypeScript version: 5.7.1

Please only submit bug reports when using the officially supported version.

=============
```

It's possible with TS 7 that an error is encountered before the code is able to detect the problematic version (or something with the version detection is broken). Maybe we need to double check the detection logic and ensure it works with TS 7 (and then also hard-error rather than warn, since it definitely will not work).

**Comment by kirkwaiblinger** [comment]
@typescript-eslint/triage-team

**Comment by kirkwaiblinger** [comment]
Also, FYI, when I try to `npm i` in the reproduction repo, I get:

```plaintext
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: typescript-eslint@8.61.1
npm error Found: typescript@7.0.1-rc
npm error node_modules/typescript
npm error   dev typescript@"^7.0.1-rc" from the root project
npm error
npm error Could not resolve dependency:
npm error peer typescript@">=4.8.4 <6.1.0" from typescript-eslint@8.61.1
npm error node_modules/typescript-eslint
npm error   dev typescript-eslint@"^8.61.1" from the root project
npm error
npm error Conflicting peer dependency: typescript@6.0.3
npm error node_modules/typescript
npm error   peer typescript@">=4.8.4 <6.1.0" from typescript-eslint@8.61.1
npm error   node_modules/typescript-eslint
npm error     dev typescript-eslint@"^8.61.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry this command with --force or --legacy-peer-deps to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/kirkwaiblinger/.npm/_logs/2026-06-22T21_24_14_038Z-eresolve-report.txt
npm error A complete log of this run can be found in: /Users/kirkwaiblinger/.npm/_logs/2026-06-22T21_24_14_038Z-debug-0.log
```

**Comment by bradzacher** [comment]
It doesn't show the nice error because it doesn't even finish importing TS. The stack trace shows that node crashed when importing ts v7.

As kirk said - we don't support TS v7 - they don't yet expose a nodejs api for us to consume. Nothing can be done about this.

**Comment by lukpsaxo** [comment]
If you follow the side-by-side instructions:

https://devblogs.microsoft.com/typescript/announcing-typescript-7-0-rc/#using-typescript-7.0-rc

you can have typescript 6 setup for typescript-eslint but still have a fast ts 7 "tsc" binary.
